Prefs [Task][26301] Create Time Tracking UI

toandv98 opened the pull request:
## Related Tickets

- [#Task 26301](https://edu-redmine.sun-asterisk.vn/issues/26301)

## WHAT

- Create Time Tracking UI.

## HOW

- DateRangePicker pick date range, FloatingActionButton show check in screen, RecyclerView show all time tracking.

## WHY

## Evidence (Screenshot or Video)

<div align="center"><img src="https://user-images.githubusercontent.com/44815970/89365350-9ea6f300-d6fe-11ea-95d0-26aa953c0179.jpg" width="250" height="530"/></div>

## Performance (Optional)

## Review Checklist

| Category      | View Point                                                   | Description                                                  | Expected Reviewer Answer | Reviewer1 (ToanDV) | Reviewer2 (HuyTQ)   |
| ------------- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------ | ------------------ | ------------------ |
| General       | Code should be self-explanatory.                             | Get a feel of story reading, while going through the code. Use appropriate name for variables, functions and classes. If you are taking more time to understand the code, then either code needs refactoring or at least comments have to be written to make it clear. | YES                      | <li>- [x] yes</li> | <li>- [x] yes</li> |
| General       | Is this function or class too big? If yes, is the function or class having too many responsibilities? |                                                              | NO                       | <li>- [x] no</li>  | <li>- [x] no</li>  |
| General       | Does the code follow DRY?: Do not Repeat Yourself            | Is there any redundant or duplicate code? DRY (Do not Repeat Yourself) principle: The same code should not be repeated more than twice. Consider reusable services, functions and components. Consider generic functions and classes. | YES                      | <li>- [x] yes</li> | <li>- [x] yes</li> |
| General       | Exception handling and cleanup (dispose) resources           |                                                              | YES                      | <li>- [x] yes</li> | <li>- [x] yes</li> |
| General       | Does the code build and execute correctly?                   |                                                              | YES                      | <li>- [x] yes</li> | <li>- [x] yes</li> |
| Conventions   | Does the source code have end-line errors?                   |                                                              | NO                       | <li>- [x] no</li>  | <li>- [x] no</li>  |
| Conventions   | No hard coding, use constants/configuration values           |                                                              | YES                      | <li>- [x] yes</li> | <li>- [x] yes</li> |
| Conventions   | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES                      | <li>- [x] yes</li> | <li>- [x] yes</li> |
| Redmine       | Does the ticket follow Sun* Redmine working process?         | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md | YES                      | <li>- [x] yes</li> | <li>- [x] yes</li> |
| Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |                                                              | YES                      | <li>- [x] yes</li> | <li>- [x] yes</li> |

## Notes (Optional)

*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

tuanpa-0908 commented on app/src/main/res/layout/fragment_time_tracking.xml:
```diff
@@ -1,6 +1,56 @@
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/recyclerTimeTracking"
+        android:layout_width="match_parent"
```
```suggestion
        android:layout_width="0dp"
```

tuanpa-0908 commented on app/src/main/res/layout/item_time_record.xml:
```diff
@@ -0,0 +1,164 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.cardview.widget.CardView xmlns:android="http://schemas.android.com/apk/res/android"
```
Hoàn toàn có thể dùng backgroundDrawable cho constraintlayout bên dưới mà. Không nên phân cấp 2 parent thế này đâu ô, hiệu năng sẽ giảm, FrameLayout trong trường hợp này chẳng có tác dụng gì ngoài việc bo cái góc mà drawable cũng làm được.

tuanpa-0908 commented on app/src/main/res/layout/item_time_record.xml:
```diff
@@ -0,0 +1,164 @@
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:background="@drawable/bg_date_time"
+            android:clickable="false"
+            android:focusable="false"
```
2 props này không cần thiết

tuanpa-0908 commented on app/src/main/res/layout/item_time_record.xml:
```diff
@@ -0,0 +1,164 @@
+            app:layout_constraintGuide_begin="@dimen/dp_80" />
+
+        <TextView
+            android:id="@+id/textView2"
```
Đặt tên có ý nghĩa nhé

tuanpa-0908 commented:
Pull nào có UI ô thêm screenshot cho dễ tracking nhé

toandv98 committed:
Task #26301
Create Time Tracking UI:
DateRangePicker pick date range
FloatingActionButton show check in screen
RecyclerView show all time tracking
